`nil` pointer exception in vspherevm controller on start up

/kind bug

**What steps did you take and what happened:**
While deploying a workload cluster, the vspheremachine controller throws a nil pointer exception
Logs - 
```
I0125 19:21:33.078152       1 http.go:136] controller-runtime/webhook/webhooks "msg"="wrote response" "code"=200 "reason"="" "webhook"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-vspherevm" "UID"="df208e5e-f0af-456c-85ae-b4df957a75ba" "allowed"=true
I0125 19:21:33.079235       1 vimmachine.go:363] capv-controller-manager/vspheremachine-controller/k8s-conformance-qsxoh8/k8s-conformance-kq6ok0-bxqx2 "msg"="VSphereVM already exists"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16d7ac5]

goroutine 692 [running]:
sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1.(*VSphereVM).GetObjectKind(0x0, 0x22cd380, 0x0)
	<autogenerated>:1 +0x5
sigs.k8s.io/cluster-api-provider-vsphere/pkg/services.(*VimMachineService).ReconcileNormal(0x3705f68, 0x2632558, 0xc000c15bc0, 0x2e, 0xc001144b00, 0xc00032eb40)
	/workspace/pkg/services/vimmachine.go:117 +0xa6c
sigs.k8s.io/cluster-api-provider-vsphere/controllers.machineReconciler.reconcileNormal(0xc000459d40, 0x2628eb8, 0x3705f68, 0x0, 0x0, 0x0, 0x2632558, 0xc000c15bc0, 0xc000c15bc0, 0x0, ...)
	/workspace/controllers/vspheremachine_controller.go:304 +0x1d1
sigs.k8s.io/cluster-api-provider-vsphere/controllers.machineReconciler.Reconcile(0xc000459d40, 0x2628eb8, 0x3705f68, 0x0, 0x0, 0x0, 0x26206b8, 0xc000c42ea0, 0xc000ff9ef0, 0x16, ...)
	/workspace/controllers/vspheremachine_controller.go:241 +0x907
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc00060c960, 0x26206b8, 0xc000c42e70, 0xc000ff9ef0, 0x16, 0xc00157aa80, 0x1c, 0xc000c42e00, 0x0, 0x0, ...)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.2/pkg/internal/controller/controller.go:114 +0x247
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00060c960, 0x2620610, 0xc000b24100, 0x2040460, 0xc0011fe180)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.2/pkg/internal/controller/controller.go:311 +0x305
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00060c960, 0x2620610, 0xc000b24100, 0x442300)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.2/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0xc0009f4b80, 0xc00060c960, 0x2620610, 0xc000b24100)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.2/pkg/internal/controller/controller.go:227 +0x6b
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.2/pkg/internal/controller/controller.go:223 +0x425
```
This is reproducible and happens each time I create a workload cluster. The CAPV pod restarts _once_ every time. 


**Environment:**

- Cluster-api-provider-vsphere version: `v1.0.2`
- Kubernetes version: (use `kubectl version`):  `1.20.4`

--- a/controllers/vspheremachine_controller.go
+++ b/controllers/vspheremachine_controller.go
@@ -36,11 +36,14 @@ import (
 	"sigs.k8s.io/cluster-api/util/patch"
 	"sigs.k8s.io/cluster-api/util/predicates"
 	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlbldr "sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 
@@ -133,7 +136,17 @@ func AddMachineControllerToManager(ctx *context.ControllerManagerContext, mgr ma
 		r.networkProvider = networkProvider
 	} else {
 		// Watch any VSphereVM resources owned by the controlled type.
-		builder.Watches(&source.Kind{Type: &infrav1.VSphereVM{}}, &handler.EnqueueRequestForOwner{OwnerType: controlledType, IsController: false})
+		builder.Watches(
+			&source.Kind{Type: &infrav1.VSphereVM{}},
+			&handler.EnqueueRequestForOwner{OwnerType: controlledType, IsController: false},
+			ctrlbldr.WithPredicates(predicate.Funcs{
+				// ignore creation events since this controller is responsible for
+				// the creation of the type.
+				CreateFunc: func(e event.CreateEvent) bool {
+					return false
+				},
+			}),
+		)
 	}
 
 	c, err := builder.Build(r)

--- a/pkg/services/vimmachine.go
+++ b/pkg/services/vimmachine.go
@@ -121,8 +121,8 @@ func (v *VimMachineService) ReconcileNormal(c context.MachineContext) (bool, err
 	}
 
 	vm, err := v.createOrPatchVSPhereVM(ctx, vsphereVM)
-
-	if err != nil && !apierrors.IsAlreadyExists(err) {
+	if err != nil {
+		ctx.Logger.Error(err, "error creating or patching VM", "vsphereVM", vsphereVM)
 		return false, err
 	}
 

